### PR TITLE
ENH: Allow to plot weighted KDEs.

### LINKED ISSUE #59087: ENH: Allow to plot weighted KDEs.
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

The current implementation does not currently allow to plot weighted KDEs.

### Feature Description

Estimation of the PDF is currently done via [scipy.stats.gaussian_kde](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gaussian_kde.html#scipy.stats.gaussian_kde) which allows for a parameter `weights`.
[pandas.DataFrame.plot.kde](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.plot.kde.html) should accept this parameter as well.

### Alternative Solutions

[Here](https://github.com/pandas-dev/pandas/blob/d9cdd2ee5a58015ef6f4d15c7226110c9aab8140/pandas/plotting/_core.py#L1411-L1522) allow to pass a parameter `weights` to [`scipy.stats.gaussian_kde`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gaussian_kde.html#scipy.stats.gaussian_kde).

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by fbourgey**
- [ ] closes #59087 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -1470,6 +1471,9 @@ def kde(
             'scott', 'silverman', a scalar constant or a callable.
             If None (default), 'scott' is used.
             See :class:`scipy.stats.gaussian_kde` for more information.
+        weights : NumPy array, optional
+            Weights of datapoints. This must be the same shape as dataset.
```
Comment (yuanx749) [inline pandas/plotting/_core.py:1475]: `dataset` is a parameter of `scipy.stats.gaussian_kde`, but it is not explicitly used here. Perhaps `datapoints` is more clear.
```suggestion
            Weights of datapoints. This must be the same shape as datapoints.
```

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -1449,6 +1449,7 @@ def hist(
     def kde(
         self,
         bw_method: Literal["scott", "silverman"] | float | Callable | None = None,
+        weights: np.ndarray | None = None,
```
Comment (mroeschke) [inline pandas/plotting/_core.py:1452]: This needs to be placed after `ind`

**Review by mroeschke — CHANGES_REQUESTED** [review]
Could you add a test an whatsnew entry in `v3.0.0.rst`?

File: doc/source/whatsnew/v3.0.0.rst [hunk 3]
```diff
@@ -593,6 +593,7 @@ Period
 
 Plotting
 ^^^^^^^^
+- Added missing parameter ``weights`` in :meth:`DataFrame.plot.kde` for the estimation of the PDF (:issue:`59337`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:596]: Can you add this under `Other Enhancements`?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @fbourgey

**Pull request merged**